### DISCUSSION TITLE
Can some one help me i am trying to make the sound and music options scroll by one

### DISCUSSION THREAD

**Discussion by Raptorq1** [Techokami/SonicWorldsNext#199] [Q&A]
Can some one help me i am trying to make the sound and music options scroll by one 

Oh and i made it a options menu Its seperate

[SoundManager.zip](https://github.com/user-attachments/files/17165235/SoundManager.zip)

**Comment by r6915ee** [comment]
I am a tad bit confused. How come you would need this? I'm just asking, since most of the time it is generally up to what the player decides to be using both the already provided options and their external volume combined, so I don't really know why you would need this in the first place.

**Reply by Raptorq1** [reply]
I need help fixing it because the options menu isnt part of the pause menu

**Reply by r6915ee** [reply]
I mean the inclusion of making the options scroll by one. I'm not exactly sure why you would need that functionality when it's technically already implemented for you.

**Reply by Raptorq1** [reply]
This is what i meant 

https://github.com/user-attachments/assets/fe91c399-75c7-48b4-ad50-987cf952b52e

This is from the base engine 


This is from my fan game


https://github.com/user-attachments/assets/6fd3fc9c-382e-4602-ad20-ea8012a77acb

**Reply by r6915ee** [reply]
Yes, it just confuses me why you would need to make it scroll by one in the first place.

**Reply by NeonSRB2** [reply]
You're lucky you know that-
I had to push all this crap into a table so I could have only 20 markers for cc
I can't blame you for having this issue honestly, since the solution swn uses right now is actually really hard to use.
So here's cc's table, it wont "scroll by one" but mania itself uses only 20 markers anyway, so it's good enough.
```
var dbtable = [
-40,
-37.700005,
-35.400009,
-33.100014,
-30.800018,
-28.500023,
-26.200027,
-23.900032,
-21.600037,
-19.300041,
-17.000046,
-14.700047,
-12.400047,
-10.100047,
-7.800047,
-5.500047,
-3.200047,
-0.900046,
1.399954,
3.699954,
5.999954
]

# the following is what you need to set to a setting on here, again, the table is only 20 entries long, VALUE is 0-20
# the code isn't litterally meant to be here btw, replace VALUE with something else perhaps and put these in functions when setting the value of that setting

AudioServer.set_bus_volume_db(AudioServer.get_bus_index("SFX"), dbtable[VALUE]) # this is for sound effects
AudioServer.set_bus_volume_db(AudioServer.get_bus_index("Music"), dbtable[VALUE]) # music

# remember to make sure you call Global.save_settings() once the menu closes to make sure these changes save!
```

**Reply by NeonSRB2** [reply]
Oh, and swn uses a seperate beep sound node to respond to the music slider's volume, just make sure your node uses the Music bus in the inspector and it should work the same

**Reply by Raptorq1** [reply]
Thanks for the help @NeonSRB2
